web/src/app/alerts/CreateAlertDialog/StepContent/Step1.js

```diff
@@ -0,0 +1,187 @@
+import React, { useRef } from 'react'
+import {
+  Grid,
+  TextField,
+  InputAdornment,
+  List,
+  ListItem,
+  ListItemText,
+  ListItemIcon,
+  Paper,
+  Typography,
+  InputLabel,
+  Chip,
+} from '@material-ui/core'
+import { makeStyles } from '@material-ui/core/styles'
+import { FormField } from '../../../forms'
+import ServiceLabelFilterContainer from '../../../services/ServiceLabelFilterContainer'
+import { Search as SearchIcon } from '@material-ui/icons'
+import FavoriteIcon from '@material-ui/icons/Star'
+import { ServiceChip } from '../../../util/Chips'
+import AddIcon from '@material-ui/icons/Add'
+
+const useStyles = makeStyles(theme => ({
+  addAll: {
+    backgroundColor: theme.palette.primary['400'],
+  },
+  chipContainer: {
+    padding: theme.spacing(0.5),
+    marginBottom: theme.spacing(2),
+    height: '9em',
+    overflow: 'auto',
+    border: '1px solid #bdbdbd',
+  },
+  endAdornment: {
+    display: 'flex',
+    alignItems: 'center',
+  },
+  noticeText: {
+    width: '100%',
+    textAlign: 'center',
+    alignSelf: 'center',
+    lineHeight: '9em',
+  },
+}))
+
+const CREATE_ALERT_LIMIT = 35
+
+export default props => {
+  const { formFields, queriedServices } = props
+
+  const fieldRef = useRef()
+  const classes = useStyles()
+
+  const labelKey = formFields.searchQuery.split(/(!=|=)/)[0]
+  const labelValue = formFields.searchQuery
+    .split(/(!=|=)/)
+    .slice(2)
+    .join('')
+
+  const AddAll = () => (
+    <Chip
+      className={classes.addAll}
+      color='primary' // for white text
+      component='button'
+      label='Add All'
+      size='small'
+      icon={<AddIcon fontSize='small' />}
+      onClick={() => {
+        const toAdd = queriedServices.map(s => s.id)
+
+        // build newState
+        let newState = formFields.selectedServices
+        toAdd.forEach(s => {
+          if (newState.length < CREATE_ALERT_LIMIT) {
+            newState = newState.concat(s)
+          }
+        })
+
+        props.onChange({ selectedServices: newState })
+      }}
+    />
+  )
+
+  const serviceChips = formFields.selectedServices.map((id, key) => {
+    return (
+      <ServiceChip
+        key={key}
```

Comment:
Use `id` as the key instead of the index. That allows React to delete just the singe DOM element of a removed service chip, instead of updating them all when one is removed.